<?xml version="1.0" encoding="UTF-8"?>
<beans xmlns="http://www.springframework.org/schema/beans" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xmlns:context="http://www.springframework.org/schema/context" xmlns:util="http://www.springframework.org/schema/util"
  xmlns:p="http://www.springframework.org/schema/p" xmlns:c="http://www.springframework.org/schema/c" xmlns:mvc="http://www.springframework.org/schema/mvc"
  xsi:schemaLocation="http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans.xsd
                           http://www.springframework.org/schema/mvc http://www.springframework.org/schema/mvc/spring-mvc.xsd
                           http://www.springframework.org/schema/context http://www.springframework.org/schema/context/spring-context.xsd
                           http://www.springframework.org/schema/util http://www.springframework.org/schema/util/spring-util.xsd"
                           
       default-init-method="initialize"
       default-destroy-method="destroy">

  <!--
    Simple Spring MVC setup for extension servlets. 
   -->

  <bean class="org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerMapping" p:order="1" />
  
  <bean class="org.springframework.web.servlet.handler.SimpleUrlHandlerMapping" p:order="2">
    <property name="mappings">
      <props>
        <prop key="/RaiseError">shibboleth.RaiseErrorController</prop>
      </props>
    </property>
  </bean>

  <bean class="org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter" />
  <bean class="org.springframework.web.servlet.mvc.SimpleControllerHandlerAdapter" />

  <bean id="shibboleth.RaiseErrorController" class="net.shibboleth.ext.spring.error.ErrorRaisingController" />

  <bean id="shibboleth.ExceptionResolver" class="net.shibboleth.ext.spring.error.ExtendedMappingExceptionResolver"
    p:defaultErrorView="%{idp.errors.defaultView:error}" p:excludedExceptions="#{getObject('%{idp.errors.excludedExceptions:undefined}')}"
    p:exceptionMappings="#{getObject('%{idp.errors.exceptionMappings:undefined}')}" />

  <bean id="RethrowingFlowExecutionExceptionHandler" class="net.shibboleth.idp.profile.impl.RethrowingFlowExecutionExceptionHandler" />

  <bean id="shibboleth.InternalViewResolver" class="org.springframework.web.servlet.view.InternalResourceViewResolver">
    <property name="order" value="1" />
    <property name="viewClass" value="org.springframework.web.servlet.view.JstlView" />
    <property name="prefix" value="/WEB-INF/jsp/" />
    <property name="suffix" value=".jsp" />
    <property name="contentType" value="text/html;charset=utf-8" />
  </bean>

  <mvc:interceptors> 
  
    <bean class="org.springframework.web.servlet.mvc.WebContentInterceptor"
      p:cacheSeconds="0" 
      p:useExpiresHeader="true" 
      p:useCacheControlHeader="true"
      p:useCacheControlNoStore="true"
      p:alwaysMustRevalidate="true" />

  </mvc:interceptors>
  
</beans>
